pep-9999.rst

```diff
@@ -0,0 +1,214 @@
+               "1.2.3"
+           ],
+           "withdrawn": "2022-06-28T16:39:06Z"
+       }
+   }
+
+
+For this field, the key (vulnerability ID) MUST be unique, and MUST take the
+form of the `OSV ID field
+<https://ossf.github.io/osv-schema/#id-modified-fields>`__. The vulnerability
+ID prefix MUST be ``PYSEC-``.
+
+The dictionary corresponding to that key MUST take the form of the `Open Source
+Vulnerabilty format (v. 1.3.1) <https://ossf.github.io/osv-schema/>`__.
+Specifically, the following fields MUST be included:
+
+- ``aliases``: `A list of IDs of the same vulnerability in other databases <https://ossf.github.io/osv-schema/#aliases-field>`__.
+- ``details``: `Additional English textual details about the vulnerability <https://ossf.github.io/osv-schema/#summary-details-fields>`__.
+- ``summary``: `A one-line, English textual summary of the vulnerability <https://ossf.github.io/osv-schema/#summary-details-fields>`__.
+- ``withdrawn`` `The time the entry should be considered to have been withdrawn <https://ossf.github.io/osv-schema/#withdrawn-field>`__.
+
+All other fields are optional. All fields that return lists MUST return empty
+lists if no value is present. All fields that return strings MUST return
+``null`` if no value is present.
+
+Additionally, a field ``fixed_in`` MUST be included. This field represents a
+list of versions for which the given vulnerability is considered fixed, and is
+an aggregate of the values of the ``fixed`` field for all events included in
+the ``affected[].ranges[].events`` field of the `OSV specification
+<https://ossf.github.io/osv-schema/#affectedrangesevents-fields>`__ for the
+specific affected version in question.
+
+An example for how to derive this from the schema format follows:
+
+.. code-block:: python
+
+    fixed_in = [
+        event["fixed"]
+        for affected in schema["affected"]
+        if current_version in affected["versions"]
+        for range in affected["ranges"]
+        for event in range["events"]
+        if "fixed" in event
+    ]
+
+
+Versions listed in the ``fixed_in`` field MUST be present in the ``versions``
+key defined in :pep:`700` and MUST have at least one corresponding file in the
+``files`` key defined in :pep:`691`.
+
+All of the vulnerability IDs listed in the ``vulnerabilities`` field MUST be
+associated with one or more of the files in the ``files`` key via the
+``vulnerability-ids`` field (see below). If there are no known vulnerabilities
+for a given project, the ``vulnerabilities`` field MUST be an empty mapping.
+
+
+Additional file information
+---------------------------
+
+One new key is added to the ``files`` key:
+
+- ``vulnerability-ids``: This field is mandatory. If a given file has known
+  vulnerabilities associated with it, it MUST contain a list of vulnerability
+  IDs, all of which MUST correspond to keys in the ``vulnerabilties`` top level
+  field. If the file has no known vulnerabilities, this field should be an
+  empty list.
```

Comment:
Making sure I understand: is the idea here that we're precomputing relevant vulnerabilities for users based on specific installable distributions, or is it to enable the equivalent of OSV's `ecosystem_specific` (e.g. "foopackage 1.2.3 is vulnerable, but only in wheel format")?

Reply:
The responses here are essentially a numeration of all files for a project, and there needs to be some association between a file and a vulnerability, hence this field.

In theory, this might mean that different distributions for the same release _could_ have different `vulnerability-ids`, and this PEP doesn't rule that out as a possibility. But in practice PyPI/OSV doesn't currently offer that level of granularity (but may in the future).

Reply:
Makes sense, thanks for clarifying!